api/v1alpha4/azuremachine_types.go

```diff
@@ -108,6 +108,9 @@ type AzureMachineSpec struct {
 	// SecurityProfile specifies the Security profile settings for a virtual machine.
 	// +optional
 	SecurityProfile *SecurityProfile `json:"securityProfile,omitempty"`
+
+	// SubnetName selects the Subnet where the VMSS will be placed
+	SubnetName string `json:"subnetName"`
```

Comment:
you also need to restore SubnetName for AzureMachine in conversion (similar to what you did for AzureMachinePool here https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/1411/files#diff-70d3cd943c26a06094c8d25a793905ab070c916f0b88f3edac84fd485843374dR49)